src/redis-helper.ts

```diff
@@ -111,25 +168,41 @@ export async function getArtistsForFestival(redisClient: redis.RedisClient, fest
 
 export async function getTopTracksForArtist(redisClient: redis.RedisClient,
                                             artist: SpotifyArtist,
-                                            tracksPerArtist: number): Promise<SpotifyTrack[]> {
+                                            tracksPerArtist: number,
+                                            allowFetchBasedOnUpdatedDate: boolean = false): Promise<SpotifyTrack[]> {
     // Coerce to a number since it'll never evaluate to true when checking if we've reached it if it's a string
     tracksPerArtist = Number(tracksPerArtist);
 
     // We need two lists because there's a chance that some tracks come from redis and some from spotify
     const topTracksFromRedis: RedisTrack[]     = [];
     let   topTracksFromSpotify: SpotifyTrack[] = [];
 
-    if (!artist.top_track_ids || artist.top_track_ids.length === 0) {
-        // We've never gotten tracks and saved their ids for this artist, need to call spotify for tracks,
-        // save their ids for this artist, and save the tracks themselves. We cache all track IDs per artist
-        // and tracks themselves, but only return the number requested
-        console.log(`No top track ids for spotify artist ${
-            artist.id}, getting from spot, saving to redis artist, and saving each track`);
+    const lastUpdated = await getArtistTopTracksLastUpdated(redisClient, artist.id);
+
+    if (!lastUpdated) {
+        console.log(`No last updated date for top tracks for artist ${artist.id} found`);
+    }
+
+    const refreshThresholdMs = new Date(Date.now() - constants.daysInMsBeforeArtistTracksRefresh);
+
+    if (!artist.top_track_ids || artist.top_track_ids.length === 0 ||
+        (allowFetchBasedOnUpdatedDate && (!lastUpdated || lastUpdated < refreshThresholdMs))) {
+        // We've never gotten tracks and saved their ids for this artist, or haven't done it for a while, so need to
+        // call spotify for tracks, save their ids for this artist, and save the tracks themselves. We cache all track
+        // IDs per artist and tracks themselves, but only return the number requested
+        console.log(`No/out of date top track ids for spotify artist ${
+            artist.id}, getting from spot, saving to redis artist, and saving each track. lastUpdated: ${
+            lastUpdated ? lastUpdated.toISOString() : 'never'}`);
         const spotifyTracks: SpotifyTrack[] = await spotifyHelper.getAllTracksForArtist(artist);
 
+        const nowISOString = new Date().toISOString();
+
         // Get and save track IDs for this artist
         redisClient.hmset(`artist:${artist.id}`,
-                          {top_track_ids : JSON.stringify(spotifyTracks.map(x => x.id))},
+                          {
+                              top_track_ids : JSON.stringify(spotifyTracks.map(x => x.id)),
+                              top_track_ids_last_updated : JSON.stringify(nowISOString)
```

Comment:
only need to stringify non-string data structures like lists or nested objects
```suggestion
                              top_track_ids_last_updated : nowISOString
```